### ISSUE TITLE
`NNXMeta.add_axis` / `remove_axis` are no-ops, silently dropping axis metadata under Linen `scan` / `vmap`

### ISSUE THREAD

**Issue by ajitashwath** [google/flax#5589]
# Description

## System information

- **OS Platform and Distribution:** Windows 11 (code read; repro not executed locally: see note under Steps to reproduce)
- **Flax, jax, jaxlib versions:** Flax at main commit `4f42ed06` (version `0.12.9` per `flax/version.py`); local env had JAX/jaxlib `0.6.2`, which cannot import NNX (`jax.experimental.hijax` missing), so behavior claims below are from code reading, not execution
- **Python version:** 3.11
- **GPU/TPU model and memory:** CPU-only environment
- **CUDA version:** N/A

## Problem you have encountered

`NNXMeta` (`flax/nnx/bridge/variables.py:56-62`) implements the `AxisMetadata` axis-update methods as stubs that silently return `self`:

```python
def add_axis(self, index: int, params: dict[Any, Any]) -> 'NNXMeta[A]':
  # TODO: implement this, supporting hooks
  return self

def remove_axis(self, index: int, params: dict[Any, Any]) -> 'NNXMeta[A]':
  # TODO: implement this, supporting hooks
  return self
```

This violates the contract on the abstract methods (`flax/core/meta.py:88-129`), which requires the metadata to be updated and `add_axis`/`remove_axis` to act as each other's inverse. The reference implementation `Partitioned.add_axis/remove_axis` (`flax/core/meta.py:275-287`) does update its names.

The stubs are reachable through `meta.add_axis` / `meta.remove_axis` (`meta.py:149-156`), called from:

- `flax/core/lift.py:791, 861` (`vmap`)
- `flax/core/lift.py:1042, 1052` (`scan`)
- `flax/linen/transforms.py:2150-2153`

`NNXMeta` is produced exactly when a bridged NNX variable carries sharding or hook metadata (`to_linen_var`, `flax/nnx/bridge/variables.py:89-99`).

**Net effect:** running a Linen `scan`/`vmap` over a `ToLinen`-wrapped module with annotated NNX variables silently keeps stale axis metadata.

## What you expected to happen

`NNXMeta.add_axis` / `remove_axis` should update the wrapped metadata (and round-trip the `on_add_axis` / `on_remove_axis` hooks), so axis metadata stays correct through Linen transforms — as `Partitioned` does.

The existing hook-dispatch machinery is already in place:

- `Variable.add_axis` / `remove_axis` (`flax/nnx/variablelib.py:1662-1678`)
- Reachable via `NNXMeta.to_nnx_variable()` (`variables.py:71-72`)

One mapping to decide: Linen passes `params: dict` (e.g. `PARTITION_NAME`), while NNX takes `(axis_index, axis_name)`.

## Logs, error messages, etc.

No error is raised: that is the bug. The failure mode is silent: stale metadata rather than an exception.

## Steps to reproduce

Whenever possible, please provide a minimal example. Please consider submitting it as a Colab link.

> **Note:** I could not execute this locally (my env's JAX `0.6.2` predates `hijax`, so `flax.nnx` won't import). Posting as a sketch for a maintainer/contributor with a current env to confirm:

```python
# Sketch (UNRUN: please verify):
# 1. Build an nnx.Module with a sharded variable (e.g. via
#    nnx.with_partitioning), wrap with nnx.bridge.ToLinen.
# 2. Apply linen scan/vmap with metadata_params over it.
# 3. Inspect the NNXMeta metadata before/after.
# Expected: metadata reflects the added/removed axis.
# Actual (per code reading): metadata object returned unchanged.
```

## Workaround tried

None available short of avoiding sharded/hooked NNX variables inside Linen-transformed `ToLinen` modules, or re-applying the metadata manually after the transform.

## Related prior work

`#4194` (2024 bridge restructuring: different scope, no overlap); no open PR addresses this (checked current open-PR list).
